tests/meshcop/commissioner.hpp

```diff
@@ -0,0 +1,309 @@
+#include "mbedtls/config.h"
+#else
+#include MBEDTLS_CONFIG_FILE
+#endif
+#include <mbedtls/net_sockets.h>
+#include <mbedtls/debug.h>
+#include <mbedtls/ssl.h>
+#include <mbedtls/entropy.h>
+#include <mbedtls/ctr_drbg.h>
+#include <mbedtls/ecjpake.h>
+#include <mbedtls/error.h>
+#include <mbedtls/certs.h>
+#include <mbedtls/timing.h>
+
+#include "agent/coap.hpp"
+#include "agent/dtls.hpp"
+#include "agent/uris.hpp"
+#include "common/tlv.hpp"
+#include "common/code_utils.hpp"
+#include "common/logging.hpp"
+#include "utils/hex.hpp"
+#include "utils/steeringdata.hpp"
+#include "web/pskc-generator/pskc.hpp"
+
+
+using namespace ot;
+using namespace ot::Utils;
+using namespace ot::BorderRouter;
+
+
+#define MBEDTLS_DEBUG_C
+#define DEBUG_LEVEL 1
+
+#define SYSLOG_IDENT "otbr-commissioner"
+
+#include "commissioner_argcargv.hpp"
+
+/**
+ * Constants
+ */
+enum
+{
+    /* max size of a network packet */
+    kSizeMaxPacket = 1500,
+
+    /* Default size of steering data */
+    kSteeringDefaultLength = 15,
+
+    /* how long is an EUI64 in bytes */
+    kEui64Len = (64 / 8),
+
+    /* how long is a PSKd in bytes */
+    kPSKdLength = 32,
+
+    /* What port does our internal server use? */
+    kPortJoinerSession = 49192,
+
+    /* 64bit xpanid length in bytes */
+    kXpanidLength = (64 / 8), /* 64bits */
+
+    /* specification: 8.10.4 */
+    kNetworkNameLenMax = 16,
+
+    /* Spec is not specific about this items max length, so we choose 64 */
+    kBorderRouterPassPhraseLen = 64,
+
+};
+
+
+
+/**
+ * Commissioner State
+ */
+enum
+{
+    kStateInvalid,
+    kStateConnected,
+    kStateAccepted,
+    kStateReady,
+    kStateAuthenticated,
+    kStateFinalized,
+    kStateDone,
+    kStateError,
+};
+
+
+/**
+ * Commissioner TLV types
+ */
+enum
+{
+    kJoinerDtlsEncapsulation
+};
+
+
+/**
+ * Commissioner Context.
+ */
+struct Context
+{
+    /** Set to true if we should commission the device */
+    bool commission_device;
+
+    /** coap instance to talk to agent & device */
+    Coap::Agent *mCoap;
+
+    /** dtls info for talking to agent & joiner */
+    Dtls::Server *mDtlsServer;
+
+    /* the session info for agent & joiner */
+    Dtls::Session *mSession;
+
+    /* dtls context information */
+    mbedtls_ssl_context *mSsl;
+    mbedtls_net_context *mNet;
+
+    /* Socket we are using with the agent */
+    int mSocket;
+
+    /* this generates our coap tokens */
+    uint16_t mCoapToken;
+
+    /* this is the commissioner session id */
+    uint16_t mCommissionerSessionId;
+
+    /** all things for the border agent */
+    struct br_agent
+    {
+
+        /** network port, in form mbed likes it (ascii) */
+        char mPort_ascii[ 7 ];
+
+        /** network address, in form mbed likes it (ascii) */
+        char mAddress_ascii[64];
+
+        /** xpanid from command line & binary form */
+        /* Note: xpanid is used to calculate the PSKc */
+        struct
+        {
+            char    ascii[  (kXpanidLength * 2) + 1 ];
```

Comment:
Saving 1 flag - but making it inconsistent with others - is not an improvement.

If/when the commissioner code is improved - why don't we address these things then.

At this point I am running out of time - I leave for vacation tomorrow and this will not be addressed for the next 3 weeks.